lnwallet/channel.go

```diff
@@ -3198,15 +3199,28 @@ func (lc *LightningChannel) validateCommitmentSanity(theirLogCounter,
 	// Calculate the commitment fee, and subtract it from the
 	// initiator's balance.
 	commitFee := feePerKw.FeeForWeight(commitWeight)
+	commitFeeMsat := lnwire.NewMSatFromSatoshis(commitFee)
+
+	// Before we apply the fee, we'll check to see if it's even payable
+	// using the initiator's balance. If the fee is greater than their
+	// current balance, then this commitment would consume their entire
+	// output, meaning the reservation would be violated.
+	switch {
+	case lc.channelState.IsInitiator && commitFee > ourBalance.ToSatoshis():
+		return ErrBelowChanReserve
+	case !lc.channelState.IsInitiator && commitFee > theirBalance.ToSatoshis():
+		return ErrBelowChanReserve
+	}
+
+	// If the above check passes, then it's safe to apply the fee.
 	if lc.channelState.IsInitiator {
-		ourBalance -= lnwire.NewMSatFromSatoshis(commitFee)
+		ourBalance -= commitFeeMsat
 	} else {
-		theirBalance -= lnwire.NewMSatFromSatoshis(commitFee)
+		theirBalance -= commitFeeMsat
 	}
 
-	// If the added HTLCs will decrease the balance, make sure
-	// they won't dip the local and remote balances below the
-	// channel reserves.
+	// If the added HTLCs will decrease the balance, make sure they won't
+	// dip the local and remote balances below the channel reserves.
 	if ourBalance < ourInitialBalance &&
```

Comment:
Similarly, this check should be done before subtracting the fee, as we allow the fee to consume value below the channel reserve.

Reply:
Fixed!